python_modules/dagster/dagster/_core/instance/persist_run_for_production.py

```diff
@@ -0,0 +1,71 @@
+from typing import Mapping, Optional
+
+from dagster._core.definitions.utils import validate_tags
+from dagster._core.host_representation.repository_location import RepositoryLocation
+from dagster._core.host_representation.selector import PipelineSelector
+from dagster._core.instance import DagsterInstance
+from dagster._core.storage.pipeline_run import DagsterRun, DagsterRunStatus
+from dagster._core.utils import make_new_run_id
+
+
+def persist_run_for_production(
+    *,
+    instance: DagsterInstance,
+    repository_location: RepositoryLocation,
+    pipeline_selector: PipelineSelector,
+    run_config: Mapping[str, object],
+    context_specific_tags: Mapping[str, str],
+    explicit_mode: Optional[str],
```

Comment:
Can this default to None?

Reply:
So I can. However I do think we rely on optional arguments too much in the codebase and I think it was one of the reasons why this got so unruly. We just kept adding "optional" parameters that really weren't optional because we didn't want to update all call sites